internal/boxcli/push.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2023 Jetpack Technologies Inc and contributors. All rights reserved.
+// Use of this source code is governed by the license in the LICENSE file.
+
+package boxcli
+
+import (
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"go.jetpack.io/devbox"
+	"go.jetpack.io/devbox/internal/boxcli/usererr"
+	"go.jetpack.io/devbox/internal/pullbox/git"
+)
+
+type pushCmdFlags struct {
+	config configFlags
+	force  bool
+}
+
+func pushCmd() *cobra.Command {
+	flags := pushCmdFlags{}
+	cmd := &cobra.Command{
+		Use:     "push",
+		Short:   "Push a config to a git repo",
+		Long:    "Push a config to a git repo. This will create a commit if needed and push it to the remote.",
+		PreRunE: ensureNixInstalled,
+		Args:    cobra.ExactArgs(0),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return pushCmdFunc(cmd, flags)
+		},
+	}
+
+	cmd.Flags().BoolVarP(
+		&flags.force, "force", "f", false,
+		"Force push even if the remote has diverged",
+	)
+
+	flags.config.register(cmd)
+
+	return cmd
+}
+
+func pushCmdFunc(cmd *cobra.Command, flags pushCmdFlags) error {
+	box, err := devbox.Open(flags.config.path, cmd.ErrOrStderr())
```

Comment:
how is `flags.config.path` set for global configs?

Reply:
the global parent command does some hackiness and basically sets the path for all subcommands in `setGlobalConfigForDelegatedCommands`